### DISCUSSION TITLE
Encrypt exsting voulme

### DISCUSSION THREAD

**Discussion by alimoezzi** [longhorn/longhorn#9309] [General]
Hi,
I have some existing node that I want to enable encryption on them.
My question is according to [here](https://github.com/longhorn/longhorn/issues/440#issuecomment-953847631), can I change the config map, add secret to enable encryption on existing volumes?

**Comment by mantissahz** [comment]
> can I change the config map, add secret to enable encryption on existing volumes?

I'm afraid that It is not possible to encrypt an existing volume without data loss because Longhorn does not handle the scenario now.

**Reply by alimoezzi** [reply]
Do you know what happens to existing volumes and new replicas after updating encryption of the storage classes?

**Reply by mantissahz** [reply]
>Do you know what happens to existing volumes and new replicas after updating encryption of the storage classes?

The existing volumes will do nothing after updating the storage class used to create volumes.
The storage class is only used when creating volumes.

**Reply by alimoezzi** [reply]
So new replicas will be encrypted while older one stays the same, right?

**Reply by mantissahz** [reply]
>So new replicas will be encrypted while older one stays the same, right?

If you mean the new replicas of existing volumes, no, the new replicas would not be encrypted (Spec. of existing volumes will not be changed after the storage class is updated.)

**Reply by alimoezzi** [reply]
Thanks, so this would be a feature request.
